code/modules/interaction_tests/test_puppeteer.dm

```diff
@@ -0,0 +1,49 @@
+/datum/test_puppeteer
+	var/mob/living/carbon/puppet
+	var/list/tracked_atoms
+
+/datum/test_puppeteer/New(carbon_type, turf/initial_location)
+	if(!initial_location)
+		initial_location = locate(179, 136, 1) // Center of admin testing area
+	puppet = new carbon_type(initial_location)
+	var/datum/mind/new_mind = new("interaction_test_[rand(1, 999999)]")
+	new_mind.transfer_to(puppet)
+
+/datum/test_puppeteer/proc/spawn_obj_in_hand(obj_type)
+	var/obj/new_obj = new obj_type(null)
+	puppet.put_in_hands(new_obj)
```

Comment:
It's possible this could fail, no? Especially if their hands are full or something.

Reply:
Yup. I've made it a test failure if this fails for now, since we're assuming the test writer knows roughly the state of things, and something failing to work here is either the test writer's mistake or an external error.